state-chain/runtime/src/chainflip/simulate_swap.rs

```diff
@@ -0,0 +1,232 @@
+// Copyright 2025 Chainflip Labs GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+use cf_chains::{
+	assets::any::ForeignChainAndAsset,
+	instances::{
+		ArbitrumInstance, AssethubInstance, BitcoinInstance, EthereumInstance, PolkadotInstance,
+		SolanaInstance,
+	},
+};
+use cf_primitives::{
+	AccountId, Asset, AssetAmount, BasisPoints, Beneficiary, DcaParameters, IngressOrEgress,
+};
+use cf_traits::OrderId;
+use pallet_cf_swapping::{BatchExecutionError, FeeType, NetworkFeeTracker, Swap};
+use sp_runtime::DispatchError;
+
+use crate::{
+	runtime_apis::{
+		CcmData, DispatchErrorWithMessage, FeeTypes, SimulateSwapAdditionalOrder,
+		SimulatedSwapInformation,
+	},
+	LiquidityPools, Runtime, Swapping,
+};
+
+use sp_runtime::traits::UniqueSaturatedInto;
+use sp_std::{collections::btree_set::BTreeSet, vec, vec::Vec};
+
+pub fn simulate_swap(
```

Comment:
Would be nice to document here what this function does and what its current limitations are.